src/main/java/productDependency/CreateDependentCategoryPage.java

```diff
@@ -0,0 +1,974 @@
+		log.info("Click on Add Price Button");
+		Assert.assertTrue(clickAddPriceButton.isDisplayed());
+		clickAddPriceButton.click();
+		//navigateBottom();
+
+	}
+	
+	@FindBy(how = How.XPATH, using = "//*[@id='dependency']/div[1]/a/span")
+	private WebElement clickDependenciesPlus;
+
+	/**
+	 * Method to click on Dependencies plus icon.
+	 * 
+	 * @throws IOException
+	 */
+	public void clickDependenciesPlus() throws IOException {
+		CreateDependentCategoryPage sp = new CreateDependentCategoryPage(driver);
+		log.info("click on Dependencies plus icon");
+		Assert.assertTrue(clickDependenciesPlus.isDisplayed());
+		clickDependenciesPlus.click();
+		JavaScriptExec.sleep();
+	}
+
+	private WebElement selectProductCategoryDependency;
+
+	/**
+	 * Method to select product category in dependencies section.
+	 * 
+	 * @throws IOException
+	 */
+	public void selectProductCategoryDependency() throws IOException {
+		CreateDependentCategoryPage sp = new CreateDependentCategoryPage(driver);
+		WebElement PCelement = driver.findElement(By.xpath("//select[@name='product.dependencyItemTypes']"));
+		Select se = new Select(PCelement);
+		se.selectByVisibleText(BasePage.getCellData(xlsxName, sheetName, 3, 0));
+
+	}
+
+	private WebElement selectProductDependency1;
+
+	/**
+	 * Method to select product in dependencies section.
+	 * 
+	 * @throws IOException
+	 */
+	public void selectProductDependency1() throws IOException {
+		CreateDependentCategoryPage sp = new CreateDependentCategoryPage(driver);
+		JavaScriptExec.sleep();
+		driver.findElement(By.xpath("//*[@id='product.dependencyItems']/option[4]")).click();
+		//*[@id="product.dependencyItems"]/option[3]
+		
+		
+		
+
+	}
+
+	private WebElement selectProductPeriod;
+
+	/**
+	 * Method to select product period in dependencies section.
+	 * 
+	 * @throws IOException
+	 */
+	public void selectProductPeriod() throws IOException {
+		CreateDependentCategoryPage sp = new CreateDependentCategoryPage(driver);
+		WebElement Productelement = driver.findElement(By.xpath("//select[@name='period']"));
+		Select se = new Select(Productelement);
+		se.selectByVisibleText(BasePage.getCellData(xlsxName, sheetName, 7, 0));
+
+	}
+	
+	@FindBy(how = How.XPATH, using = "//input[@id='product.dependencyMin']")
+	
+	//*[@id="product.dependencyMin"]
+	private WebElement enterMin;
+
+	/**
+	 * Method to Enter Price Rate.
+	 * 
+	 * @throws IOException
+	 */
+	public void enterMin() throws IOException {
+		CreateDependentCategoryPage sp = new CreateDependentCategoryPage(driver);
+		log.info("Enter Min value");
+		JavaScriptExec.sleep();
+		WaitClass.WaitForElementisDisplay(driver, 10, enterMin);
+		Assert.assertTrue(enterPriceRate.isDisplayed());
+		//enterPriceRate1.clear();
+		enterMin.clear();
+		enterMin.sendKeys("1");
+
+}
+	@FindBy(how = How.XPATH, using = "//*[@id='dependency']/div[2]/div/table/tbody/tr/td[6]/a/img")
+	private WebElement clickDependencyPlusIcon;
+
```

Comment:
@mayankchhajer Please define the name of the variable correctly, this WebElement is to add DependencyLine. Please Correct this.

Reply:
Done